### ISSUE TITLE
Sending Data to PubMed

### ISSUE THREAD

**Issue by Melissa37** [elifesciences/elife-pubmed-xml-generation#33]
@gnott can you let me know where you currently get this from and how?
For instance, if we stop publishing the element-citation details for datasets within the data availability statement and put them in the reference with all other dataset references, how will this affect the workflow?

**Comment by gnott** [comment]
Assuming I understand the question correctly, and I follow backwards the steps followed to add the datasets:

It starts at the `elife-tools` XML parser's `datasets_json()` function https://github.com/elifesciences/elife-tools/blob/develop/elifetools/parseJATS.py#L3757). This looks for `<sec>` in the `<back>` with a particular `type`, and then parses `<element-citation>` and data availability into the format the API and journal uses.

The data is used in the `elife-article` project to populate the `Article` object with data, in the `parse.py` module's `build_datasets()` function (https://github.com/elifesciences/elife-article/blob/develop/elifearticle/parse.py#L105).

If you were to move the data, we could adapt the parsers and logic to the new structure, provided they are still labelled as `"generated"` and `"used"` datasets.

We would want to check the results against the API schema for validity too, which could be part of the round of changes in the `elife-tools` project's `parseJATS.py` parser.

**Comment by Melissa37** [comment]
Thanks @gnott perfect! So right now it looks for `<element-citation>` with "generated" in `<sec>` in `<back>` and converts these to PubMed data, so if we asked it to look for "generated" in `<element-citation>` just in `<back>` it would pick these up from references in the reference list. 

Can I leave this open as we'll need to make this change at some point? I don't think it needs to wait for when we swap in production because it would then be able to handle data citations in Data Availibility AND those in the main reference list.

**Comment by gnott** [comment]
We can most likely support more than one XML data structure for the datasets. Ergo, the coding can be done before you supply the new format. But, in order for me to get started, if you could please supply a good sample of the XML to parse, which could be included in code tests.

I will add this to our project board into your column and you can let me know when I should look at it again.

**Comment by Melissa37** [comment]
Fab, thanks. Will do.

**Comment by Melissa37** [comment]
Hi @gnott I've read this again more sensibly and it would be great to slot this in if you have time - it would be useful for the tool for other publishers if they ever use stuff we built(!) and we will make this change at some point in the future!

However, for re-use I think it would need to rely on the publication-type="data"

**Comment by gnott** [comment]
I'm taking a fresh look at this issue now you've mentioned it again @Melissa37. I think the PubMed deposits care not about whether a dataset is "generated" or "used", and all the datasets are included in the deposit. A possible enhancement to this is it could also get all the references of type "data" from the article data and include those in the list of datasets in the deposit. Am I on the right track there, or have I missed something?

The datasets XML itself may we should discuss how they're specifeid as "generated" or "used", because that may have implications on how the data is displayed on the journal. Maybe if you are able to point me to an example XML of how the dataset would be specified in the citation list, then I could see where to check for the "generated" or "used" value.

**Comment by gnott** [comment]
I'm also looking at the Crossref depositing logic, and I think there the "generated" and "used" status has a greater importance, which indicates whether a the dataset is `references` or `isSupplementedBy` type in Crossref's schema. There too it could probably also be configured to additionally look at the citation list, but these "generated" and "used" attributes will be important.

**Comment by Melissa37** [comment]
>I'm taking a fresh look at this issue now you've mentioned it again @Melissa37. I think the PubMed deposits care not about whether a dataset is "generated" or "used", and all the datasets are included in the deposit. A possible enhancement to this is it could also get all the references of type "data" from the article data and include those in the list of datasets in the deposit. Am I on the right track there, or have I missed something?

Yes on the right track! However, I thought PubMed only wanted to receive those from their controlled list

>The datasets XML itself may we should discuss how they're specifeid as "generated" or "used", because that may have implications on how the data is displayed on the journal. Maybe if you are able to point me to an example XML of how the dataset would be specified in the citation list, then I could see where to check for the "generated" or "used" value.

This is for Crossref deposits - they want to know which they are when we send to them - that's in the code right? (For Crossref)?
We don't have current plans to indicate that in the journal itself once we move these references to the reference list, but it might be a nice feature (which product would be involved in designing the UI) so I'd like to keep that distinction in the XML. HOWEVER, it would greatly reduce our author effort if we stopped insisting they add datasets into the table at submission, and that's the only way we get that info. But again, it's an open question - if we stopped asking them to add this data in the submission system, would we see a drop off of Data citations being provided and would that create more work for us.

@acollings For Info/thoughts

**Comment by acollings** [comment]
Thanks Graham and Melissa,

For now I think it's better to keep asking for the datasets during submission, especially if we want to keep the distinction between newly generated and previously published ones somewhere.

Cheers, Andy

**Comment by gnott** [comment]
> This is for Crossref deposits - they want to know which they are when we send to them - that's in the code right? (For Crossref)?

Yes, one part is the Crossref deposits depends on the `generated` and `used` values to specify the right type of link in the metadata.

It is also used in the JSON schema, but now I think I'm understanding that in future all data sources would be in the citation list, and there would be no more listed under the `Data availability` section on the journal article page. I guess the data availability statement would still be there, though @Melissa37?

**Comment by Melissa37** [comment]
>It is also used in the JSON schema, but now I think I'm understanding that in future all data sources would be in the citation list, and there would be no more listed under the Data availability section on the journal article page. I guess the data availability statement would still be there, though @Melissa37?

Yes, that is correct. We'll still collect the datasets in EJP, which give us the distinction of whether they are generated or used, which we'll retain in the XML, but this will be in the reference list, not as major datasets listed under the DAS. Does that make sense?

**Comment by gnott** [comment]
Code was merged into the PubMed library last Friday to enhance the data deposits as planned in this issue, if I got it all correct.

I'm deploying this updated PubMed to the production environment `elife-bot` instance now.

I will close this particular issue, with the hopes it was all done correctly. We may want to spot-check one in the future if an article XML has some data citations that are not listed in the data sources section.
